flight/Modules/Stabilization/stabilization.c

```diff
@@ -423,18 +423,21 @@ static void stabilizationTask(void* parameters)
 					if (reinit)
 						pids[PID_RATE_ROLL + i].iAccumulator = 0;
 
-					if(fabs(stabDesiredAxis[i]) > max_axislock_rate) {
+					if (fabsf(stabDesiredAxis[i]) > max_axislock_rate) {
 						// While getting strong commands act like rate mode
-						rateDesiredAxis[i] = stabDesiredAxis[i];
+						rateDesiredAxis[i] = bound_sym(stabDesiredAxis[i], settings.ManualRate[i]);
+
+						// Reset accumulator
 						axis_lock_accum[i] = 0;
 					} else {
-						// For weaker commands or no command simply attitude lock (almost) on no gyro change
+						// For weaker commands or no command simply lock (almost) on no gyro change
 						axis_lock_accum[i] += (stabDesiredAxis[i] - gyro_filtered[i]) * dT;
 						axis_lock_accum[i] = bound_sym(axis_lock_accum[i], max_axis_lock);
-						rateDesiredAxis[i] = pid_apply(&pids[PID_ATT_ROLL + i], axis_lock_accum[i], dT);
-					}
 
-					rateDesiredAxis[i] = bound_sym(rateDesiredAxis[i], settings.MaximumRate[i]);
+						// Compute the inner loop
+						float tmpRateDesired = pid_apply(&pids[PID_ATT_ROLL + i], axis_lock_accum[i], dT);
```

Comment:
why define tmp variable here rather than just assign rateDesiredAxis[i] on this line and then bound it on the next line? not wrong but more consistent style with code above.